packages/components/organisms/f-checkout/src/components/Selector.vue

```diff
@@ -104,11 +110,41 @@ export default {
             this.selectedAvailableFulfilmentTime = selectedFulfilmentTime;
             this.updateChangedField('orderTime');
 
+            this.setAsapFlag(selectedFulfilmentTime);
+
             // TODO - Update to use different from/to times when the API supports it
             this.updateFulfilmentTime({
                 from: selectedFulfilmentTime,
                 to: selectedFulfilmentTime
             });
+        },
+
+        /**
+         * Initialise the first set of times so if the user proceeds with the preset delivery time
+         * we have access to the from/to times up front.
+         *
+         * @param times
+         */
+        initFulfilmentTime (times) {
+            if (times.length && times[0].value) {
+                this.updateFulfilmentTime({
+                    from: times[0].value,
+                    to: times[0].value
+                });
+            }
+        },
+
+        /**
+         * Set asap to `true` if the availableFulfilment time matches the selected fulfilment time,
+         * otherwise set it to false.
+         *
+         * @param selectedFulfilmentTime
+         */
+        setAsapFlag (selectedFulfilmentTime) {
+            const isAsapSelected = this.availableFulfilment.isAsapAvailable
+                && this.availableFulfilment.times[0].from === selectedFulfilmentTime;
```

Comment:
do all intermediate properties in `this.availableFulfilment.times[0].from` always exist? For example can there be a situation when `this.availableFulfilment.times.length` is 0?

Reply:
This method would only be triggered when the user manually changes the time they want their delivery (via the delivery drop down). So `times` would exist.